SDAppDelegate.m

```diff
@@ -413,42 +385,93 @@ - (void) tableView:(NSTableView *)aTableView willDisplayCell:(id)aCell forTableC
 /******************************************************************************/
 
+dispatch_source_t CreateDebounceDispatchTimer(double debounceTime, dispatch_queue_t queue, dispatch_block_t block) {
+    dispatch_source_t timer = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, queue);
+    
+    if (timer) {
+        dispatch_source_set_timer(timer, dispatch_time(DISPATCH_TIME_NOW, debounceTime * NSEC_PER_SEC), DISPATCH_TIME_FOREVER, (1ull * NSEC_PER_SEC) / 10);
+        dispatch_source_set_event_handler(timer, block);
+        dispatch_resume(timer);
+    }
+    
+    return timer;
+}
+
+- (void) runQueryDebounced:(NSString*)query {
+    if (self.debounceTimer != nil) {
+            dispatch_source_cancel(self.debounceTimer);
+            self.debounceTimer = nil;
+        }
+        
+        dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+        double secondsToThrottle = .300f;
+        self.debounceTimer = CreateDebounceDispatchTimer(secondsToThrottle, queue, ^{
+            [self runQuery:query];
+            
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [self.listTableView reloadData];
+            });
+        });
+}
+
 - (void) runQuery:(NSString*)query {
     query = [query lowercaseString];
+    
+    NSMutableString * combinedStuff = [NSMutableString string];
 
-    self.filteredSortedChoices = [self.choices mutableCopy];
+    [self.choices enumerateObjectsUsingBlock:^(SDChoice *obj, NSUInteger idx, BOOL *stop) {
+        [combinedStuff appendString:[NSString stringWithFormat:@"%@\n", obj.raw]];
+    }];
+    [combinedStuff deleteCharactersInRange:NSMakeRange([combinedStuff length]-1, 1)];
 
-    // analyze (cache)
-    for (SDChoice* choice in self.filteredSortedChoices)
+    NSMutableArray* fzfResult = [self executeFzfOnOptions: combinedStuff fzfQuery:query];
+    self.filteredSortedChoices = [NSMutableArray arrayWithCapacity:[fzfResult count]];
+    
+    for (NSString *someString in fzfResult) {
+        if([someString length] == 0) continue;
+        
+        SDChoice* choice = [[SDChoice alloc] initWithString:someString];
+        [self.filteredSortedChoices addObject:choice];
         [choice analyze: query];
+        [choice render];
+    }
 
-    if ([query length] >= 1) {
+    self.choice = 0;
+    [self reflectChoice];
+}
 
-        // filter out non-matches
-        for (SDChoice* choice in [self.filteredSortedChoices copy]) {
-            if (!choice.hasAllCharacters)
-                [self.filteredSortedChoices removeObject: choice];
-        }
+- (NSMutableArray *)executeFzfOnOptions:(NSString *)options fzfQuery:(NSString *)query
+{
+    NSTask *task = [[NSTask alloc] init];
+    [task setLaunchPath:@"/bin/zsh"];
+    
+    NSArray *arguments = [NSArray arrayWithObjects: @"-c",
+                          [NSString stringWithFormat:@"echo \"%@\" | fzf --reverse --exact --filter \"%@\"", options, query], nil];
```

Comment:
remove please `--exact` that is what I mention in comment
https://github.com/chipsenkbeil/choose/pull/31#issuecomment-1339433965

or even better make that string as option that can be passed to application during execution (with default fallback) - that way each user can customize in any way